### ISSUE TITLE
e621 updated, now not working

### ISSUE THREAD

**Issue by 99spooner** [Chirmaya/BooruSlideshow#80]
first off i already know that the original dev isn't working on this anymore but im hoping someone in this community could help.

e621 just updated its site and unless i hit a wrong button or something booru slideshow no longer shows it as an active site. does anyone know if there's any way too fix it or is  it pretty much dead. if its the latter would anyone know any alternatives for e621

**Comment by Bobmester** [comment]
I believe they've just updated their API https://e621.net/help/api

Now, I'm an amateur and too lazy to fix it, but not too lazy to try to answer your question! It shouldn't be difficult to fix, an amateur could do it, it's just a matter of finding the motivation to do it instead of just doing without it.

**Comment by Bobmester** [comment]
I believe they've just updated their API https://e621.net/help/api

Now, I'm an amateur and too lazy to fix it, but not too lazy to try to answer your question! It shouldn't be difficult to fix, an amateur could do it, it's just a matter of finding the motivation to do it instead of just doing without it.

**Comment by DontTalkToMeThx** [comment]
Check out my fork here: https://github.com/DontTalkToMeThx/BooruSlideshow You'll have to go into developer mode and manually pack the extension for it to work though.

**Comment by 99spooner** [comment]
@DontTalkToMeThx thanks that works good but i was seeing if it could be fixed on the firefox version (if that's even possible)

**Comment by DontTalkToMeThx** [comment]
@99spooner Of course, I'll try but firefox seems to have some sort of verification system. However you can still pack it yourself by going to about:debugging#/runtime/this-firefox and load temporary add-on then select the zip file in my fork it should work, but I think you'd need to do that every time, so I'll see if I can get it verified.

If you are going to use the temporary add-on method, do know I just redid the zip file, so if you downloaded it before now, you'd have to download it again.

**Comment by 99spooner** [comment]
@DontTalkToMeThx actually im curious if there is anyway to make it into an xpi file. the reason is because i actually use a Firefox alternate browser (waterfox to be specific) that allows me to install unverified add-ons just by dragging the xpi file

if not possible thats fine too and i will wait to see if you can get it verified or i will just use the chrome one

**Comment by DontTalkToMeThx** [comment]
@99spooner An XPI file is just an archive, like zips. All you have to do is change the extension, it should work. I'll link a download to it https://easyupload.io/sfmpo9

**Comment by DontTalkToMeThx** [comment]
Just for an update as well. I'm working on favorite list slideshow and filtering

**Comment by DontTalkToMeThx** [comment]
@99spooner I'm bundling it into an electron app, not much was needed to be changed, but I didn't have time to test everything. I'm going to post that for you so you don't have to worry about any firefox signing. You'll run it like any other desktop app, but you need to choose your platform first (I only tested it on windows 64 bit).

The platforms are
- linux arm64
- linux armv7l
- linux ia32
- linux x64
- win32-arm64
- win32 ia32
- win32 x64 (this is windows 64 bit)

I'll push the changes to my fork soon, then you can clone the repo and run the appropriate executable. But because I only have windows, I can only test it on windows, so if it doesn't work on linux I can't really do much about it.

**Comment by 99spooner** [comment]
@DontTalkToMeThx I have win32 x64

**Comment by DontTalkToMeThx** [comment]
@99spooner Check out [this](https://cdn.is-just-a.dev/booruslideshowelectron-win32-x64.zip) the app is called booruslideshowelectron.exe don't move it, keep it in the file and create a shortcut for it if you want to move it. Additionally I can help you transfer your favorites list, but it will require some manual code editing, but it isn't that difficult and I'll walk you through it if you want.

**Comment by 99spooner** [comment]
@DontTalkToMeThx it seems to work good so far, also dont really need to add favorites as really didnt have any to begin with (probably less then 10 or even 5).

**Comment by DontTalkToMeThx** [comment]
@99spooner Yep, if you run into any issues, make an issue [on my fork](https://github.com/DontTalkToMeThx/BooruSlideshow) and mention you're using the electron app

**Comment by DontTalkToMeThx** [comment]
It seems I had an oversight in data loading. I'm going to update the app, most of it works, but and false values are not retained. (It just happens to be most values are false, so it's not that noticeable). Within an hour of me posting this, it will be updated [here](https://cdn.is-just-a.dev/booruslideshowelectron-win32-x64.zip) make sure to clear your cache for it before redownloading it. I'll update this comment when the file is live

**Comment by 99spooner** [comment]
@DontTalkToMeThx im not sure if im missing something but i cant seem to find the issues tab on the fork, it may be because im not that knowledgeable with githubs menus but im not sure

**Comment by DontTalkToMeThx** [comment]
@99spooner Yeah that's my fault. Issues aren't enabled automatically on forks, but I just enabled them. https://github.com/DontTalkToMeThx/BooruSlideshow/issues heres a link

**Comment by glowl5** [comment]
@DontTalkToMeThx hey, could you say where does the extension store favourites?

**Comment by DontTalkToMeThx** [comment]
@glowl5 Sure, it's a little complicated because it's not saved in a file, it's saved in chrome's local storage, however we can download it to a file by doing this:
1. Open the extension and press Ctrl + Shift + i or F12 if that doesnt work.
2. Navigate to "console" on the top menu
3. Paste this: 
```
chrome.storage.local.get(['personalListItems'],
function (data) {
    var blob = new Blob([JSON.stringify(data.personalListItems)], {type: 'text/json'}),
        e    = document.createEvent('MouseEvents'),
        a    = document.createElement('a')

    a.download = "listItems.json"
    a.href = window.URL.createObjectURL(blob)
    a.dataset.downloadurl =  ['text/json', a.download, a.href].join(':')
    e.initMouseEvent('click', true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null)
    a.dispatchEvent(e)
})
```

And save the file. Depending on how big your favorites list is, it may look weird, but that's not something I can easily fix.

**Comment by glowl5** [comment]
@DontTalkToMeThx incredibly helpful, thanks a bunch ^^
